supplementary/alica_ros_turtlesim/ros2/src/base.cpp

```diff
@@ -38,6 +43,10 @@ Base::Base(rclcpp::Node::SharedPtr nh, const std::string& name, const int agent_
     ac->setLogger<alicaRosLogger::AlicaRosLogger>(agent_id);
 
     spinThread = std::thread([this]() { spinner.spin(); });
+
+    LockedBlackboardRW bb(ac->editGlobalBlackboard());
+    bb.set<std::shared_ptr<turtlesim::TurtleInterfaces>>("turtle", std::make_shared<turtlesim::TurtleRos2Interfaces>(name));
+    bb.set("spawned", false);
```

Comment:
we want do it _after_ creating external thread?

\* _after_ is more like about order in code, we don't have any gurarntees when thread will be created

Reply:
The order doesnt matter in this case, whats important is that before we intiialize the alicaContext, we set the value in the blackboard. Only after that plans and behaviours will be executed and they need acess to this blackboard value.